test/test_linalg.py

```diff
@@ -1868,6 +1869,266 @@ def test_nuclear_norm_exceptions_old(self, device):
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(*floating_types())
+    def test_svd_tall_some(self, device, dtype):
+        self._test_svd_helper((20, 5), True, False, device, dtype)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(*floating_types())
+    def test_svd_tall_all(self, device, dtype):
+        self._test_svd_helper((20, 5), False, False, device, dtype)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(*floating_types())
+    def test_svd_tall_some_col_maj(self, device, dtype):
+        self._test_svd_helper((5, 20), True, True, device, dtype)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(*floating_types())
+    def test_svd_tall_all_col_maj(self, device, dtype):
+        self._test_svd_helper((5, 20), False, True, device, dtype)
+
+    # ~~~ tests for torch.linalg.svd ~~~
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_linalg_svd_compute_uv(self, device, dtype):
+        """
+        Test the default case, compute_uv=True. Here we have the very same behavior as
+        numpy
+        """
+        t = torch.randn((10, 11), device=device, dtype=dtype)
+        np_t = t.cpu().numpy()
+        for full_matrices in (True, False):
+            # check linalg.svd vs numpy
+            expected = np.linalg.svd(np_t, full_matrices, compute_uv=True)
+            actual = torch.linalg.svd(t, full_matrices, compute_uv=True)
+            self.assertEqual(actual, expected)
+            # check linalg.svd vs linalg.svd(out=...)
+            out = (torch.empty_like(actual[0]),
+                   torch.empty_like(actual[1]),
+                   torch.empty_like(actual[2]))
+            out2 = torch.linalg.svd(t, full_matrices, compute_uv=True, out=out)
+            self.assertEqual(actual, out)
+            self.assertEqual(actual, out2)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_linalg_svd_no_compute_uv(self, device, dtype):
+        """
+        Test the compute_uv=False case. Here we have a different return type than
+        numpy: numpy returns S, we return (empty, S, empty)
+        """
+        t = torch.randn((10, 11), device=device, dtype=dtype)
+        np_t = t.cpu().numpy()
+
+        def is_empty(x):
+            return x.numel() == 0 and x.dtype == t.dtype and x.device == t.device
+
+        for full_matrices in (True, False):
+            # check linalg.svd vs numpy
+            np_s = np.linalg.svd(np_t, full_matrices, compute_uv=False)
+            USV = torch.linalg.svd(t, full_matrices, compute_uv=False)
+            assert is_empty(USV.U)
+            self.assertEqual(USV.S, np_s)
+            assert is_empty(USV.V)
+            # check linalg.svd vs linalg.svd(out=...)
+            out = (torch.Tensor(), torch.empty_like(USV.S), torch.Tensor())
```

Comment:
Don't use torch.Tensor(), which is deprecated. Instead use torch.tensor/torch.empty/torch.empty_like. 

Does this mean that an empty CPU tensor is an acceptable out= value when running on CUDA? torch.linalg.svd should verify that the tensors passed to out= are valid whether compute_uv is True or False.

Reply:
> Don't use torch.Tensor(), which is deprecated. Instead use torch.tensor/torch.empty/torch.empty_like.
ea0aca4b09

> Does this mean that an empty CPU tensor is an acceptable out= value when running on CUDA? torch.linalg.svd should verify that the tensors passed to out= are valid whether compute_uv is True or False.

currently yes, but that's true for most (if not all) operators defined `BatchLinearAlgebra.cpp`. E.g. `torch.eig`, `torch.svd`, `torch.solve`, and in general all operators which uses the pattern `out.resize(X).copy_(...)`.

Reply:
Yikes. I think you're right and that's a bug. Let's get this case correct and I've updated https://github.com/pytorch/pytorch/issues/49468 to include a bullet point for testing this behavior.

Reply:
It looks to me that the best way to implement this behavior would be to add this extra check to `resize_output`, and make sure to use it everywhere.
I'm happy to work on this but I would prefer to do this in a separate PR, since it probably involves touching a lot of different code, introduce new tests, etc.